### PR TITLE
Update documents of lock view for sprint 4

### PR DESCRIPTION
**Pull request by MyonKeminta**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

The feature Lock View has changed much in this sprint. Update documents for it.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@MyonKeminta Please involve a technical review.

File: information-schema/information-schema-data-lock-waits.md [hunk 1]
```diff
@@ -22,24 +18,50 @@ DESC data_lock_waits;
...
+
+* `"db_id"`：该 key 所属的数据库（schema）的 ID。
+* `"db_name"`：该 key 所属的数据库（schema）的名称。
+* `"table_id"`：该 key 所属的表的 ID。
+* `"table_name"`：该 key 所属的表的名称。
+* `"handle_type"`：该 row key 的 handle 类型，其可能的值有：
+    * `"int"`：handle 为 int 类型，即 handle 为 row id；
+    * `"common"`：非 int64 类型的 handle，在启用 clustered index 时非 int 类型的主键会显示为此类型；
+    * `"unknown"`：当前暂不支持的 handle 类型。
+* `"partition_handle"`：是否为 partition handle。
```
Comment (MyonKeminta) [inline information-schema/information-schema-data-lock-waits.md:58]: Here's some misunderstanding of mine about the `partition_handle` field. I'll fix it next week.

File: functions-and-operators/tidb-functions.md [hunk 2]
```diff
@@ -259,3 +259,72 @@ Check Table Before Drop: false
...
+```sql
++---------------------------------------+
+| tidb_decode_sql_digests(@digests, 10) |
++---------------------------------------+
+| ["begin",null,"select * f..."]        |
++---------------------------------------+
+1 row in set (0.01 sec)
+```
+
+上述调用指定了第二个参数（即阶段长度）为 10，而查询结果中的第三条语句的长度大于 10，因而仅保留了前 10 个字符，并在尾部添加了 `"..."` 表示发生了截断。
```
Comment (qiancai) [inline functions-and-operators/tidb-functions.md:321]:
```suggestion
上述调用指定了第二个参数（即截断长度）为 10，而查询结果中的第三条语句的长度大于 10，因而仅保留了前 10 个字符，并在尾部添加了 `"..."` 表示发生了截断。
```

File: troubleshoot-lock-conflicts.md [hunk 3]
```diff
@@ -266,34 +266,15 @@ select * from information_schema.deadlocks;
...
-
-```sql
-+-------------+----------------------------+--------------------+--------------------+-----------------------------------------+
-| deadlock_id | occur_time                 | try_lock_trx_id    | trx_holding_lock   | digest_text                             |
-+-------------+----------------------------+--------------------+--------------------+-----------------------------------------+
-|           1 | 2021-06-04 08:22:38.765699 | 425405959304904707 | 425405959304904708 | update `t` set `v` = ? where `id` = ? ; |
-|           1 | 2021-06-04 08:22:38.765699 | 425405959304904708 | 425405959304904707 | update `t` set `v` = ? where `id` = ? ; |
-+-------------+----------------------------+--------------------+--------------------+-----------------------------------------+
-```
+查询结果会显示死锁错误中多个事务之间的等待关系和各个事务当前正在执行的 SQL 语句的归一化形式（即去掉参数和格式的形式），以及发生冲突的 key 及其从 key 中解读出的一些信息。
```
Comment (qiancai) [inline troubleshoot-lock-conflicts.md:277]: 这里可能有些简略，是否可以再详细说明下呢？比如告诉用户从  `CURRENT_SQL_DIGEST_TEXT ` 列可以获取各个事务当前正在执行的 SQL 语句的归一化形式（即去掉参数和格式的形式）

File: troubleshoot-lock-conflicts.md [hunk 4]
```diff
@@ -363,25 +344,29 @@ ba07e3cc34b6b3be7b7c2de7fe9]
...
+       MEM_BUFFER_BYTES: 19
+             SESSION_ID: 113
+                   USER: root
+                     DB: test
+        ALL_SQL_DIGESTS: ["0fdc781f19da1c6078c9de7eadef8a307889c001e05f107847bee4cfc8f3cdf3","a4e28cc182bdd18288e2a34180499b9404cd0ba07e3cc34b6b3be7b7c2de7fe9","06da614b93e62713bd282d4685fc5b88d688337f36e88fe55871726ce0eb80d7"]
+                   sqls: ["begin ;","select * from `t` where `id` = ? for update ;","update `t` set `v` = `v` + ? where `id` = ? ;"]
 1 row in set (0.01 sec)
 ```
 
+上述查询中，对 `CLUSTER_TIDB_TRX` 表的 `ALL_SQL_DIGESTS` 列使用了 [`TIDB_DECODE_SQL_DIGESTS`](/functions-and-operators/tidb-functions.md#tidb_decode_sql_digests) 函数，以尝试将该列（内容为一组 SQL Digest）转换为其对应的 SQL 语句，以便于阅读。
```
Comment (qiancai) [inline troubleshoot-lock-conflicts.md:370]: “以尝试将该列（内容为一组 SQL Digest）转换为其对应的 SQL 语句“，这里请问是“转换为其对应的 SQL 语句的归一化形式”吗？

**Review by qiancai — COMMENTED** [review]
@MyonKeminta PTAL

File: information-schema/information-schema-data-lock-waits.md [hunk 5]
```diff
@@ -22,24 +18,50 @@ DESC data_lock_waits;
...
+> * 来自不同 TiKV 节点的信息不能保证是同一时间点的快照。
+> * `SQL_DIGEST` 列中的信息（SQL Digest）为 SQL 语句进行归一化后计算得到的 hash。`SQL_DIGEST_TEXT` 列中的信息为内部从 Statements Summary 系列表中查询得到，因而存在内部查询不到对应语句的可能性。关于 SQL Digest 和 Statements Summary 相关表的详细说明，请参阅[Statement Summary Tables](/statement-summary-tables.md)。
+
+## `KEY_INFO`
+
+`KEY_INFO` 列中展示了对 `KEY` 列中所给出的 key 的详细信息，以 JSON 格式给出。其包含的信息如下：
+
+* `"db_id"`：该 key 所属的数据库（schema）的 ID。
+* `"db_name"`：该 key 所属的数据库（schema）的名称。
+* `"table_id"`：该 key 所属的表的 ID。
```
Comment (cfzjywxk) [inline information-schema/information-schema-data-lock-waits.md:52]: May need to explain if it's a partitioned table, if it's the partition table id or the real physical table id

File: functions-and-operators/tidb-functions.md [hunk 6]
```diff
@@ -259,3 +259,72 @@ Check Table Before Drop: false
 ### MySQL 兼容性
 
 `TIDB_VERSION` 是 TiDB 特有的函数，和 MySQL 不兼容。如果要求兼容 MySQL，可以使用 `VERSION` 获取版本信息，但结果不包含详细的构建信息。
+
+## TIDB_DECODE_SQL_DIGESTS
+
+`TIDB_DECODE_SQL_DIGESTS` 函数用于在集群中查询一组 SQL Digest 所对应的 SQL 语句的归一化形式（即去除格式和参数后的形式）。函数接受 1 个或 2 个参数：
```
Comment (cfzjywxk) [inline functions-and-operators/tidb-functions.md:265]: The `归一化` seems a bit confusing， how about `规范化` or `模板化`.

File: functions-and-operators/tidb-functions.md [hunk 7]
```diff
@@ -259,3 +259,72 @@ Check Table Before Drop: false
...
+* `digests`：字符串类型，该参数应符合 JSON 字符串数组的格式，数组中的每个字符串应为一个 SQL Digest。
+* `stmtTruncateLength`：可选参数，整数类型，用来限制返回结果中每条 SQL 语句的长度，超过指定的长度会被截断。0 表示不限制长度。
+
+返回一个字符串，符合 JSON 字符串数组的格式，数组中的第 *i* 项为参数 `digests` 中的第 *i* 个元素所对应的语句。如果参数 `digests` 中的某一项不是一个有效的 SQL Digest 或系统无法查询到其对应的 SQL 语句，则返回结果中对应项为 `null`。如果指定了截断长度（`stmtTruncateLength > 0`），则返回结果中每条超过该长度的语句，保留前 `stmtTruncateLength` 个字符，并在尾部增加 `"..."` 后缀表示发生了截断。如果参数 `digests` 为 `NULL`，则函数的返回值为 `NULL`。
+
+> **注意：**
+>
+> * 仅持有 [PROCESS](https://dev.mysql.com/doc/refman/8.0/en/privileges-provided.html#priv_process) 权限的用户可以使用该函数。
+> * `TIDB_DECODE_SQL_DIGESTS` 执行时，内部从 Statement Summary 一系列表中查询每个 SQL Digest 所对应的语句，因而并不能保证对任意 SQL Digest 都总是能查询到对应的语句，只有在集群中执行过的语句才有可能被查询到，且是否能查询到受 Statement Summary 表相关配置的影响。有关 Statement Summary 表的详细说明，参见 [Statement Summary Tables](/statement-summary-tables.md)。
+> 该函数开销较大，在行数很多的查询中（比如在规模较大、比较繁忙的集群上查询 `information_schema.cluster_tidb_trx` 全表时）直接使用该函数可能导致查询运行时间较长。请谨慎使用。
```
Comment (cfzjywxk) [inline functions-and-operators/tidb-functions.md:276]: May need to explain more about the impact and the underlying mechanism.

File: information-schema/information-schema-data-lock-waits.md [hunk 8]
```diff
@@ -22,24 +18,50 @@ DESC data_lock_waits;
...
+    * `"int"`：handle 为 int 类型，即 handle 为 row id；
+    * `"common"`：非 int64 类型的 handle，在启用 clustered index 时非 int 类型的主键会显示为此类型；
+    * `"unknown"`：当前暂不支持的 handle 类型。
+* `"partition_handle"`：是否为 partition handle。
+* `"handle_value"`：handle 的值。
+* `"index_id"`：该 index key 所属的 index id。
+* `"index_name"`：该 index key 所属的 index 名称。
+* `"index_values"`：该 index key 中的 index value。
+
+其中，不适用或当前无法查询到的信息会被省略。比如，row key 的信息中不会包含 `index_id`、`index_name` 和 `index_values`，index key 不会包含 `handle_type` 和 `partition_handle`，已经被 drop 掉的表中的 key 的信息可能只有 `table_id` 等几个 ID。
```
[the remark this answers is not in the document]
Comment (MyonKeminta) [reply information-schema/information-schema-data-lock-waits.md:64]: 没有 db_id。这个地方我想了下更确切地说应该是“无法从 schema 信息，因而不会包含 `table_name`，`db_id`，`db_name`，`index_name` 等字段，且无法区分 key 是否属于分区表“

File: functions-and-operators/tidb-functions.md (hunk 6 above)
Comment (MyonKeminta) [reply functions-and-operators/tidb-functions.md:265]: The term `归一化` comes directly from the [Statements Summary Table](https://docs.pingcap.com/zh/tidb/v5.0/statement-summary-tables#statement-summary-tables) 's document.

File: functions-and-operators/tidb-functions.md [hunk 9]
```diff
@@ -259,3 +259,72 @@ Check Table Before Drop: false
...
+
+* `digests`：字符串类型，该参数应符合 JSON 字符串数组的格式，数组中的每个字符串应为一个 SQL Digest。
+* `stmtTruncateLength`：可选参数，整数类型，用来限制返回结果中每条 SQL 语句的长度，超过指定的长度会被截断。0 表示不限制长度。
+
+返回一个字符串，符合 JSON 字符串数组的格式，数组中的第 *i* 项为参数 `digests` 中的第 *i* 个元素所对应的语句。如果参数 `digests` 中的某一项不是一个有效的 SQL Digest 或系统无法查询到其对应的 SQL 语句，则返回结果中对应项为 `null`。如果指定了截断长度（`stmtTruncateLength > 0`），则返回结果中每条超过该长度的语句，保留前 `stmtTruncateLength` 个字符，并在尾部增加 `"..."` 后缀表示发生了截断。如果参数 `digests` 为 `NULL`，则函数的返回值为 `NULL`。
+
+> **注意：**
+>
+> * 仅持有 [PROCESS](https://dev.mysql.com/doc/refman/8.0/en/privileges-provided.html#priv_process) 权限的用户可以使用该函数。
+> * `TIDB_DECODE_SQL_DIGESTS` 执行时，内部从 Statement Summary 一系列表中查询每个 SQL Digest 所对应的语句，因而并不能保证对任意 SQL Digest 都总是能查询到对应的语句，只有在集群中执行过的语句才有可能被查询到，且是否能查询到受 Statement Summary 表相关配置的影响。有关 Statement Summary 表的详细说明，参见 [Statement Summary Tables](/statement-summary-tables.md)。
```
[the remark this answers is not in the document]
Comment (MyonKeminta) [reply functions-and-operators/tidb-functions.md:275]: TiDB 内部。我改一下

File: troubleshoot-lock-conflicts.md (hunk 4 above)
Comment (MyonKeminta) [reply troubleshoot-lock-conflicts.md:370]: 是

**Comment by MyonKeminta** [comment]
@cfzjywxk @qiancai PTAL again, thanks

File: information-schema/information-schema-data-lock-waits.md [hunk 10]
```diff
@@ -51,36 +73,13 @@ select * from information_schema.data_lock_waits\G
...
+                   KEY: 7480000000000000355F728000000000000001
+              KEY_INFO: {"db_id":1,"db_name":"test","table_id":53,"table_name":"t","handle_type":"int","handle_value":"1"}
+                TRX_ID: 426790594290122753
+CURRENT_HOLDING_TRX_ID: 426790590082449409
+            SQL_DIGEST: 38b03afa5debbdf0326a014dbe5012a62c51957f1982b3093e748460f8b00821
+       SQL_DIGEST_TEXT: update `t` set `v` = `v` + ? where `id` = ?
+1 row in set (0.01 sec)
 ```
 
 以上查询结果显示，ID 为 `425405024158875649` 的事务在执行 Digest 为 `"f7530877a35ae65300c42250abd8bc731bbaf0a7cabc05dab843565230611bb22"` 的语句的过程中，试图在 `"7480000000000000355f728000000000000002"` 这个 key 上获取悲观锁，但是该 key 上的锁目前被 ID 为 `425405016242126849` 的事务持有。
```
[the remark this answers is not in the document]
Comment (qiancai) [reply information-schema/information-schema-data-lock-waits.md]: @MyonKeminta PTAL

File: troubleshoot-lock-conflicts.md (hunk 3 above)
Comment (qiancai) [reply troubleshoot-lock-conflicts.md:277]: @MyonKeminta PTAL

File: information-schema/information-schema-data-lock-waits.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (MyonKeminta) [reply information-schema/information-schema-data-lock-waits.md]: 改了

File: troubleshoot-lock-conflicts.md (hunk 3 above)
Comment (MyonKeminta) [reply troubleshoot-lock-conflicts.md:277]: 这个下面补充了一段说明

**Review by qiancai — APPROVED** [review]
LGTM

**Review by TomShawn — APPROVED** [review]

File: information-schema/information-schema-tidb-trx.md [hunk 11]
```diff
@@ -65,73 +70,83 @@ select * from information_schema.tidb_trx\G
...
+all_sql_digests: ["e6f07d43b5c21db0fbb9a31feac2dc599787763393dd5acbfad80e247eb02ad5","04fa858fa491c62d194faec2ab427261cc7998b3f1ccf8f6844febca504cb5e9","b83710fa8ab7df8504920e8569e48654f621cf828afbe7527fd003b79f48da9e"]
+       all_sqls: ["begin","insert into `t` values ( ... )","update `t` set `v` = `v` + ?"]
+*************************** 2. row ***************************
+             id: 426789921471332353
+all_sql_digests: ["e6f07d43b5c21db0fbb9a31feac2dc599787763393dd5acbfad80e247eb02ad5","38b03afa5debbdf0326a014dbe5012a62c51957f1982b3093e748460f8b00821"]
+       all_sqls: ["begin","update `t` set `v` = `v` + ? where `id` = ?"]
 ```
 
-## SQL Digest
+此查询
```
Comment (TomShawn) [inline information-schema/information-schema-tidb-trx.md:120]: @MyonKeminta 这里是否还有什么需要补充吗？

Comment (MyonKeminta) [reply information-schema/information-schema-tidb-trx.md:120]: 应该是写的时候写漏了。。我补一下

Comment (MyonKeminta) [reply information-schema/information-schema-tidb-trx.md:120]: https://github.com/pingcap/docs-cn/pull/6882

**Pull request merged**